SwiftGit2/CommitIterator.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Created by Arnon Keereena on 4/28/17.
+// Copyright (c) 2017 GitHub, Inc. All rights reserved.
+//
+
+import Result
+import libgit2
+
+public class CommitIterator: IteratorProtocol {
+    public typealias Element = Result<Commit, NSError>
+    var repo: Repository
+    var branch: Branch
+    var revisionWalker: OpaquePointer? = nil
+    var oid: git_oid
+    private var unsafeCommit: OpaquePointer? = nil
+    
+    public init(repo: Repository, branch: Branch) {
```

Comment:
Oh, there are lots of reasons to iterate over commits. Most often you'll iterate over the difference between 2 commitish objects (commit, tag, branch). This used to find:

1. New commits in a branch/PR
2. The merge base
3. The ahead/behind count of a branch

"Rev walking", as libgit2 calls it, is a core conscript for git.